### PR TITLE
Update the advanced istio installation docs

### PR DESCRIPTION
**Pull request by ReToCode**
## Motivation
The current docs about advanced istio installation 
- are a bit outdated (sidecar injection is enabled by default in recent istio)
- the structure is not really clear (e.g. `Using Istio mTLS feature` is below `Choosing an Istio installation`)

## Proposed Changes
- Restructured the docs outline
- Reordered some blocks to have a more natural flow (install, configure, then verify)
- Removed the info about enabling automatic sidecar injection

### CODE REVIEW

**Review by psschwei — COMMENTED** [review]
/approve

/assign @nak3 @skonto 
for technical review

File: docs/install/installing-istio.md [hunk 1]
```diff
@@ -22,71 +22,51 @@ You can view the latest tested Istio version on the [Knative Net Istio releases
...
+    to avoid manually injection sidecars to all `pods` in `knative-serving`.
+
+1. Enable sidecar injection on `knative-serving` system namespace.
 
     ```bash
     kubectl label namespace knative-serving istio-injection=enabled
     ```
 
-1. Set `PeerAuthentication` to `PERMISSIVE` on knative-serving system namespace
+2. Set `PeerAuthentication` to `PERMISSIVE` on knative-serving system namespace
```
Comment (nak3) [inline docs/install/installing-istio.md:69]: The number `1` should be fine as markdown automatically assign to the correct number.

File: docs/install/installing-istio.md [hunk 2]
```diff
@@ -149,54 +166,20 @@ custom: custom-local-gateway
 If there is a change in service ports (compared to that of
 `knative-local-gateway`), update the port info in the gateway accordingly.
 
-### Verifying your Istio install
 
-View the status of your Istio installation to make sure the install was
+## Verifying your Istio installation
+
+View the status of your Istio installation to make sure the installation was
```
Comment (nak3) [inline docs/install/installing-istio.md:172]: The recent `istioctl` command has `verify-install` option so shall we change the docs to use it?
(I'm also fine for `kubectl get pods --namespace`, though.)

**Review by nak3 — COMMENTED** [review]
Thank you so much! LGTM.
The change of this PR looks perfect. I added just a few additional suggestion to change.

File: docs/install/installing-istio.md [hunk 3]
```diff
@@ -100,14 +80,51 @@ by creating a YAML file using the following template:
...
     ```bash
     kubectl apply -f <filename>.yaml
     ```
     Where `<filename>` is the name of the file you created in the previous step.
 
-After you install the cluster local gateway, your service and deployment for the local gateway is named `knative-local-gateway`.
+
+## Configuring the installation
+
+### Configuring DNS
```
[the remark this answers is not in the document]
Comment (ReToCode) [reply docs/install/installing-istio.md]: Should we link that section then? If a user reads only the advanced docs, he would not see that section about DNS right?

[the remark this answers is not in the document]
Comment (nak3) [reply docs/install/installing-istio.md]: Either is fine for me.
Your point makes sense but the section does not explain how to install `net-istio` and `serving` so I think an user cannot complete the installation by reading only the advanced docs, though.

**Comment by nak3** [comment]
/lgtm
/hold

Hold for https://github.com/knative/docs/pull/5402#discussion_r1095406173 if you would like to add the link otherwise let's merge.

File: docs/install/installing-istio.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ReToCode) [reply docs/install/installing-istio.md]: You are right. Seems odd that we do not mention that here. What do you think about this: https://github.com/knative/docs/pull/5402/commits/bfc7ae6b849f4cdf3ac6f965c56973ba58fe7cab to resolve both?

[the remark this answers is not in the document]
Comment (nak3) [reply docs/install/installing-istio.md]: Awesome :star2:

**Pull request merged**